[laishuya] iP

laishuya opened the pull request:
# Luigi 🔫 
> _"Yahoo!"_  

Luigi helps you remember the tasks you have to do. It's, 
- text-based
- easy to use
- easy to download

All you need to do is,
1. download the jar file from [here](https://github.com/laishuya/ip/releases/tag/v0.1/)
2. type `java -jar luigi.jar` on your terminal to run Luigi
3. add your tasks
4. let it manage your tasks for you 🤯 

And it is **FREE**!

Features:
- [x] Add todo, deadline and event tasks
- [x] Find tasks with the same keyword or date
- [x] Ability to mark, unmark and delete tasks

> [!NOTE]
> User guide on how to navigate Luigi chatbot coming soon!

> [!WARNING]
> The main class below is still a work in progress and is not yet functional. This page will be updated when the main class is complete.

If you are a Java programmer, you can use it to practise Java. Here's the `main` method:
```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

laishuya committed:
Add greeting

laishuya committed:
Made greetUser method private

laishuya committed:
Add support for Echo

laishuya committed:
Create list to track items

laishuya committed:
Create Task class

laishuya committed:
Add mark as done feature

laishuya committed:
Add deadline, event, todo classes

laishuya committed:
Integrate todo, deadline, event into the chatbot

laishuya committed:
Perform TextUiTesting using I/O redirection

laishuya committed:
Add error handling and switch statements

laishuya committed:
Add delete command

laishuya committed:
Reorganise code to fit coding conventions

laishuya committed:
Edit .gitignore to exclude irrelevant files

laishuya committed:
Add methods to load and save file

laishuya committed:
Handle load file being corrupted

laishuya committed:
Merge branch 'branch-Level-7'

laishuya committed:
Add date and time feature to Deadline and Event classes

laishuya committed:
Add ability to print all tasks occurring on same day

laishuya committed:
Merge branch 'branch-Level-8'

laishuya committed:
Complete TaskList and Storage classes

laishuya committed:
Complete Parser, Ui, Command classes

laishuya committed:
Merge branch 'A-MoreOOP'

laishuya committed:
Merge branch 'add-gradle-support'

laishuya committed:
Add JUnit tests for TaskList and Deadline

laishuya committed:
Edit build.gradle to package app as JAR file

laishuya committed:
Add simple javadoc comment as a test

laishuya committed:
Implement find command for level 9

laishuya committed:
Merge branch 'branch-Level-9'

laishuya committed:
Fix coding style violations

laishuya committed:
Merge branch 'branch-A-CodingStandard'

laishuya committed:
Add javadoc comments to give more information about the code

laishuya committed:
Merge branch 'branch-A-JavaDoc'

laishuya committed:
Add checkstyle and use it to detect coding style violations

laishuya committed:
Merge branch 'branch-A-CheckStyle'

laishuya committed:
Set up basic GUI for Luigi Chatbot

laishuya committed:
Finish working on GUI to replace text-based UI

laishuya committed:
Merge branch 'branch-Level-10'

panomia01 commented on src/main/java/luigi/TaskList.java:
```diff
@@ -0,0 +1,182 @@
+     * @param index The position of the Task in the list.
+     * @return A string of the unmarked Task.
+     */
+    public String unmark(int index) {
```
Function name should be written in camelCase

panomia01 commented on src/main/java/luigi/commands/EventCommand.java:
```diff
@@ -0,0 +1,42 @@
+ */
+public class EventCommand extends Command {
+    private final String description;
+    private final String from;
```
perhaps a noun is more intuitive

laishuya replied:
That makes sense, thanks for the input!

panomia01 commented on src/main/java/luigi/commands/EventCommand.java:
```diff
@@ -0,0 +1,42 @@
+public class EventCommand extends Command {
+    private final String description;
+    private final String from;
+    private final String to;
```
same for this too!

panomia01 commented on src/main/java/luigi/ui/MainWindow.java:
```diff
@@ -0,0 +1,65 @@
+    public void initialize() {
+        String greeting = "Hello! I'm Luigi!\nHow can I help you?";
+        scrollPane.vvalueProperty().bind(dialogContainer.heightProperty());
+        dialogContainer.getChildren().add(DialogBox.getLuigiDialog(greeting, luigiImage));
```
Love that you have images to your chatbot

laishuya replied:
Ty! Good luck for the rest of your ip and tp :)

panomia01 reviewed (commented):
Overall, I found your code easy to read. Only have some naming conventions

HarishB99 commented on src/main/java/luigi/Luigi.java:
```diff
@@ -0,0 +1,48 @@
+public class Luigi {
+    private static String FILE_PATH = "./data/luigi.txt";
+    private Storage storage;
+    private TaskList list;
```
```suggestion
    private TaskList tasks;
```

laishuya replied:
You're right. Thanks for pointing it out!

HarishB99 commented on src/main/java/luigi/Storage.java:
```diff
@@ -0,0 +1,114 @@
+ * If the data file and directory are missing, then they will be created by the program.
+ */
+public class Storage {
+    private static ArrayList<Task> list = new ArrayList<>();
```
```suggestion
    private static ArrayList<Task> tasks = new ArrayList<>();
```

HarishB99 commented on src/main/java/luigi/TaskList.java:
```diff
@@ -0,0 +1,182 @@
+            if (task instanceof Deadline) {
+                Deadline deadline = (Deadline) task;
+                if (deadline.getLocalDate().equals(targetDate)) {
+                    sb.append(deadline + "\n");
```
```suggestion
                    sb.append(deadline + System.lineSeparator());
```

laishuya replied:
Wow, I didn't know \n is platform dependent, while System.lineSeparator() is platform independent. Thanks!

HarishB99 reviewed (commented):
In overall, I liked your code naming conventions as they were easy to read and follow. Your code also looks well organised and clean. Just a few suggestions:

1. Maybe you could use System.lineSeparator() instead of "\n"?
2. "list" might be sounding a bit too vague. I noticed in other parts of the codebase you referred to it as "tasks" or "taskList" which sounds more descriptive of what the list holds. Maybe you can try switching to the latter?

Otherwise, a job well done!

laishuya committed:
Fix coding style violations based on PR review feedback

laishuya committed:
Use assert feature to check important assumptions

TaskList class does not check if parameters are null.

It is better to check for null parameters because they will cause the
chatbot to produce incorrect functionality.

Let's,
* use the Java assert feature to check the parameter is not null
* print an error message if the parameter is null

laishuya committed:
Refactor getListToPrint to use Java streams

getListToPrint method uses a traditional for loop and StringBuilder
object.

Refactoring the method to implement Java streams makes the code more
succinct, readable and maintainable.

Let's,
* refactor getListToPrint to use streams for mapping and joining strings
* reduce boilerplate code and improve readability of getListToPrint

laishuya committed:
Fix minor coding standard violation

laishuya committed:
Refactor code in loadFile method into methods

The original loadFile method was verbose with multiple levels of
abstraction, making it hard to read.

Refactoring the code enforces the SLAP principle.

Let's,
* refactor the code that makes a new file into the makeNewFile method
* refactor the code that scans the data file into the scanContentsInFile
method

laishuya committed:
Merge pull request #1 from laishuya/branch-A-Assertions

Use assert feature to check important assumptions

laishuya committed:
Merge branch 'master' of https://github.com/laishuya/ip

laishuya committed:
Merge branch 'master' into branch-A-CodeQuality

laishuya committed:
Merge branch 'master' into branch-A-Streams

laishuya committed:
Merge pull request #2 from laishuya/branch-A-CodeQuality

Improve code quality

laishuya committed:
Merge branch 'master' of https://github.com/laishuya/ip

laishuya committed:
Merge branch 'master' into branch-A-Assertions

laishuya committed:
Merge branch 'master' into branch-A-Streams

laishuya committed:
Merge pull request #3 from laishuya/branch-A-Streams

Implement java streams in the ip

laishuya committed:
Merge branch 'master' of https://github.com/laishuya/ip

laishuya committed:
Merge branch 'master' into branch-A-CodeQuality

laishuya committed:
Merge branch 'master' into branch-A-Assertions

laishuya committed:
Fix minor code quality issues

laishuya committed:
Add reminder extension

laishuya committed:
Merge branch 'branch-BCD-Extension'

laishuya committed:
Add gradle.yml file for continuous integration

laishuya committed:
Remove IO Redirection in gradle.yml file

laishuya committed:
Fix coding style violations

laishuya committed:
Merge branch 'master' into branch-A-Assertions

laishuya committed:
Merge branch 'branch-A-Assertions'

laishuya committed:
Merge branch 'master' into branch-A-CodeQuality

laishuya committed:
Merge branch 'branch-A-CodeQuality'

laishuya committed:
Improve GUI by allowing automatic resizing and customizing colors

laishuya committed:
Add image for user guide

laishuya committed:
Update user guide

laishuya committed:
Merge branch 'master' of https://github.com/laishuya/ip

laishuya committed:
Add images of commands for the user guide

laishuya committed:
Make more changes to user guide

laishuya committed:
Fix minor issues in user guide

laishuya committed:
Add some JUnit tests

laishuya committed:
Finalise product for submission
